oe_translation.post_update.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * @file
+ * OpenEuropa Translation post update file.
+ */
+
+declare(strict_types = 1);
+
+/**
+ * Enable OE Translation Poetry Legacy module.
+ */
+function oe_translation_post_update_00001(): void {
+  // If the OE Translation Poetry module is not enabled, we bail out.
+  if (!\Drupal::service('module_handler')->moduleExists('oe_translation_poetry')) {
+    return;
+  }
+  // Enable the legacy module.
+  \Drupal::service('module_installer')->install(['oe_translation_poetry_legacy']);
+
+  // Query database to find all the node revisions that have at least a Poetry
+  // request sent.
+  $database = \Drupal::database();
+  $query = $database->query("SELECT tmgmt_job_item.item_rid,
```

Comment:
This query is not ok. You group on item_rid, achieves nothing. Try this query:

```
SELECT ji.item_rid, ji.item_id, j.poetry_request_id__code,j.poetry_request_id__year,
       j.poetry_request_id__number,j.poetry_request_id__version,
       j.poetry_request_id__part,j.poetry_request_id__product FROM tmgmt_job j
       INNER JOIN tmgmt_job_item ji ON j.tjid = ji.tjid
       WHERE j.translator = 'poetry'

       	   GROUP BY ji.item_id
       	  ORDER BY ji.item_rid DESC
```
This one simply groups on the item_id (which is the node ID) and returns unique values per node. Also, it sorts DESC by item_rid (revision ID), so that for each node, the latest job is found.